### PR TITLE
[ffigen] Expose retainAndReturnPointer and add castFromPointer to Blocks

### PR DESCRIPTION
**Pull request by liamappelbe**
`retainAndReturnPointer` was private before, and used for passing around objects in blocks. `castFromPointer` already exists for `ObjCObject`s, I'm just adding it to blocks. I found these useful when prototyping https://github.com/liamappelbe/web_kit_plugin. I didn't end up using them in the final demo, but they're good escape hatches to have anyway.

### CODE REVIEW

File: pkgs/ffigen/lib/src/code_generator/objc_built_in_functions.dart [hunk 1]
```diff
@@ -250,7 +250,7 @@ class $name implements ${w.ffiLibraryPrefix}.Finalizable {
   /// Return a pointer to this object.
   $idType get pointer => _id;
 
-  $idType _retainAndReturnId() {
+  $idType retainAndReturnPointer() {
```
Comment (dcharkes) [inline pkgs/ffigen/lib/src/code_generator/objc_built_in_functions.dart:253]: Generate a dartdoc comment on how to use this.

File: pkgs/ffigen/test/native_objc_test/block_test.dart [hunk 2]
```diff
@@ -260,6 +260,29 @@ void main() {
...
 
+    Pointer<Void> blockManualRetainRefCountTest() {
+      final block = IntBlock.fromFunction(lib, makeAdder(4000));
+      expect(lib.getBlockRetainCount(block.pointer.cast()), 1);
+      final rawBlock = block.retainAndReturnPointer().cast<Void>();
+      expect(lib.getBlockRetainCount(rawBlock.cast()), 2);
+      return rawBlock.cast();
+    }
+
+    blockManualRetainRefCountTest2(Pointer<Void> rawBlock) {
```
Comment (dcharkes) [inline pkgs/ffigen/test/native_objc_test/block_test.dart:271]: Return type missing.

(The test would be easier to read if you return the count here and have the expect in the test itself.)

**Review by dcharkes — APPROVED** [review]

File: pkgs/ffigen/lib/src/code_generator/objc_built_in_functions.dart (hunk 1 above)
Comment (liamappelbe) [reply pkgs/ffigen/lib/src/code_generator/objc_built_in_functions.dart:257]: Done

File: pkgs/ffigen/test/native_objc_test/block_test.dart (hunk 2 above)
Comment (liamappelbe) [reply pkgs/ffigen/test/native_objc_test/block_test.dart]: Done

**Pull request merged**